### ISSUE TITLE
Infinite loop / wrong parsing on a trivial arm64 binary

### ISSUE THREAD

**Issue by remyoudompheng** [binsec/binsec#21]
binsec >= 0.7 seems to fail when processing a trivial Aarch64 hello world:
```
0000000000000714 <main>:
     714: 52800020      mov     w0, #1
     718: d65f03c0      ret

Disassembly of section .fini:
```

The binary was compiled using this C source: `int main() { return 1; }`
(I am not attaching a binary program, any program seems to reproduce the issue)

Script repro.ini:
```
sp := 0x100000000

starting from 0x714

reach 0x2000
```

Command:
```
binsec -sse -sse-debug-level 1 -sse-script repro.ini -isa aarch64 -aarch64-debug-level 1 hello
```

Output:
```
[smt:info] Found Z3 in the path.
[sse:debug] Running SSE on hello
[sse:debug] Reading script from repro.ini
[aarch64:debug] Parsing (address . 0x0000000000000714)
(opcode . 0x52800020)
(size . 4)
(mnemonic . "mov        w0, #0x1")
(0x0000000000000714,0) x0<64> := 0x0000000000000001; goto 1
(0x0000000000000714,1) goto (0x0000000000000718,0)

[aarch64:debug] Parsing (address . 0x0000000000000718)
(opcode . 0x52800020)
(size . 4)
(mnemonic . "mov        w0, #0x1")
(0x0000000000000718,0) x0<64> := 0x0000000000000001; goto 1
(0x0000000000000718,1) goto (0x000000000000071c,0)

[aarch64:debug] Parsing (address . 0x000000000000071c)
(opcode . 0x52800020)
(size . 4)
(mnemonic . "mov        w0, #0x1")
(0x000000000000071c,0) x0<64> := 0x0000000000000001; goto 1
(0x000000000000071c,1) goto (0x0000000000000720,0)

<repeating endlessly the same instruction>

[aarch64:debug] Parsing (address . 0x0000000000001ffc)
(opcode . 0x52800020)
(size . 4)
(mnemonic . "mov        w0, #0x1")
(0x0000000000001ffc,0) x0<64> := 0x0000000000000001; goto 1
(0x0000000000001ffc,1) goto (0x0000000000002000,0)

[sse:warning] Cut path 0 (max depth) @ 0x00000714
[sse:info] Empty path worklist: halting ...
[sse:info] SMT queries
             Preprocessing simplifications
               total          0
               sat            0
               unsat          0
               constant enum  0
             
             Satisfiability queries
               total          0
               sat            0
               unsat          0
               unknown        0
               time           0.00
               average        -nan
             
           Exploration
             total paths                      1
             completed/cut paths              0
             pending paths                    1
             stale paths                      0
             failed assertions                0
             branching points                 0
             max path depth                   1595
             visited instructions (unrolled)  1595
             visited instructions (static)    1595
```

The version of `unisim_archisec` in the OPAM environment is 0.0.5

The problem happens with tags 0.7.1, 0.7.2, 0.7.3

With version 0.6.3 parsing is correct:
```
[aarch64:debug] Parsing (address . 0x0000000000000714)
(opcode . 0x52800020)
(size . 4)
(mnemonic . "mov        w0, #0x1")
(0x0000000000000714,0) x0<64> := 0x0000000000000001; goto 1
(0x0000000000000714,1) goto (0x0000000000000718,0)

[aarch64:debug] Parsing (address . 0x0000000000000718)
(opcode . 0xd65f03c0)
(size . 4)
(mnemonic . "ret")
(0x0000000000000718,0) %%0<64> := x30<64>; goto 1
(0x0000000000000718,1) goto (%%0<64>)

[sse:debug] Selecting path #2 (among 3)
[sse:warning] Jump (0x00000718, 1) : goto %%0<64> 
              could have led to invalid address 0x00000000; skipping
[sse:debug] Selecting path #1 (among 2)
[sse:warning] Jump (0x00000718, 1) : goto %%0<64> 
              could have led to invalid address 0x00000001; skipping
[sse:debug] Selecting path #0 (among 1)
[sse:warning] Jump (0x00000718, 1) : goto %%0<64> 
              could have led to invalid address 0x00000003; skipping
[sse:info] Empty path worklist: halting ...
...
             visited instructions (unrolled)  2
             visited instructions (static)    2
```

**Comment by recoules** [comment]
Hi @remyoudompheng,

Thank you for reporting this bug.
The problem comes from the fact that `Aarch` decoder does not update the position in the byte stream.
I will publish a fix as soon as possible.

Regards,
Frédéric

**Comment by recoules** [comment]
Hi @remyoudompheng,

The problem was small and should now be fixed in `0.7.4`.
Sorry for this inconvenience.

Regards
